crowbar_framework/app/models/node_object.rb

```diff
@@ -289,7 +291,10 @@ def availability_zone
   def availability_zone=(value)
     @node["crowbar_wall"] ||= {}
     @node["crowbar_wall"]["openstack"] ||= {}
-    @node["crowbar_wall"]["openstack"]["availability_zone"] = value
+    if @node["crowbar_wall"]["openstack"]["availability_zone"] != value
```

Comment:
@vuntz as I mentioned it in the "eliminate saves" PR earlier, this pattern might worth extracting to a helper, something like: `set_value_and_call_function_if_changed` or I believe ruby must have better ways to express it.